src/tests/ftest/launch.py

```diff
@@ -52,2294 +68,2973 @@
-    return result["status"]
+        finally:
+            # Remove the temporary remote directory on each host
+            command = f"{sudo_command}rm -fr {tmp_copy_dir}"
+            if not run_remote(logger, hosts, command).passed:
+                message = f"Error removing temporary remote copy directory {tmp_copy_dir}"
+                self._fail_test(self.result.tests[-1], "Process", message)
+                return_code = 16
 
+        return return_code
 
-def get_service_status(hosts, service):
-    """Get the status of the daos_server.service.
+    def _process_core_files(self, test_job_results):
+        """Generate a stacktrace for each core file detected.
 
-    Args:
-        hosts (NodeSet): hosts on which to get the service state
-        service (str): name of the service
+        Args:
+            test_job_results (str): the location of the core files
 
-    Returns:
-        dict: a dictionary with the following keys:
-            - "status":       status code: 0 = success, 512 = failure
-            - "stop":         NodeSet where to stop the daos_server.service
-            - "disable":      NodeSet where to disable the daos_server.service
-            - "reset-failed": NodeSet where to reset the daos_server.service
+        Returns:
+            int: status code: 0 = success, 256 = failure
 
-    """
-    status = {
-        "status": 0,
-        "stop": NodeSet(),
-        "disable": NodeSet(),
-        "reset-failed": NodeSet()}
-    status_states = {
-        "stop": ["active", "activating", "deactivating"],
-        "disable": ["active", "activating", "deactivating"],
-        "reset-failed": ["failed"]}
-    command = "systemctl is-active {}".format(service)
-    task = get_remote_output(hosts, command)
-    for output, nodelist in task.iter_buffers():
-        output_lines = [line.decode("utf-8") for line in output]
-        nodeset = NodeSet.fromlist(nodelist)
-        log(" {}: {}".format(nodeset, "\n".join(output_lines)))
-        for key, state_list in status_states.items():
-            for line in output_lines:
-                if line in state_list:
-                    status[key].add(nodeset)
-                    break
-    if task.num_timeout() > 0:
-        nodeset = NodeSet.fromlist(task.iter_keys_timeout())
-        status["status"] = 512
-        status["stop"].add(nodeset)
-        status["disable"].add(nodeset)
-        status["reset-failed"].add(nodeset)
-        log("  {}: TIMEOUT".format(nodeset))
-    return status
-
-
-def indent_text(indent, text):
-    """Prepend the specified number of spaces to the specified text.
+        """
+        core_file_processing = CoreFileProcessing(logger)
+        try:
+            error_count = core_file_processing.process_core_files(test_job_results, True)
+            if error_count:
+                message = f"Errors detected processing test core files: {error_count}"
+                self._fail_test(self.result.tests[-1], "Process", message)
```

Comment:
(pylint-no-value-for-parameter) No value for argument 'fail_reason' in staticmethod call